nav2_bt_navigator/include/nav2_bt_navigator/navigator.hpp

```diff
@@ -106,23 +106,121 @@ class NavigatorMuxer
   std::mutex mutex_;
 };
 
+class NavigatorBase
```

Comment:
Why not have the `Navigator` class that already exists here be the base for pluginlib? There's no requirement that it has to be a pure virtual class.

Reply:
This class looks almost entirely as a pass through anyway.

Reply:
![Screenshot from 2022-07-14 09-55-53](https://user-images.githubusercontent.com/60467877/178862017-ddd45668-0c31-47b7-9a0e-a80e2a6fcc9a.png)

I know that a base class with a template causes an error in the classloader.

Reply:
Its possible, you just have to be careful. Here's an example of it working: https://github.com/artofnothingness/mppic/tree/b557139c48ae6dc75a06354e9e9001781e4b96ec

In this example, they template both the base function and the output critics. That's not necessary for our case.

Reply:
Eh, but yeah the function for loading the plugins and the class loader object will need to be templated